javascript/javascript3/week2/JS3W2/js3w2.js

```diff
@@ -0,0 +1,105 @@
+// Movies exercise
+
+async function getBadMovies() {
+  try {
+    const responseMovies = await fetch(
+      "https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json"
+    );
+    const movies = await responseMovies.json();
+    const badMovies = movies.filter((movie) => movie.rating < 4);
+    const badMoviesAfter = badMovies.filter((movie) => movie.year > 2000);
+    console.log(badMovies);
+    console.log(badMoviesAfter);
+  } catch (error) {
+    throw "Something went wrong";
+  }
+}
+getBadMovies();
+
+//Promise that resolves after set time
+
+function callsAfterXSeconds(num) {
+  const miliseconds = num * 1000;
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, miliseconds);
+  });
+}
+
+callsAfterXSeconds(3)
+  .then(() => {
+    console.log(`Am I called asynchronously?`);
+  })
+  .catch((err) => {
+    console.log("Something went wrong");
+  });
+
+// Rewrite time
+
+async function setTimeoutPromise(sec) {
+  const miliseconds = sec * 1000;
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(`Called after ${sec} seconds`);
+    }, miliseconds);
+  });
+}
+
+setTimeoutPromise(8)
+  .then(console.log)
+  .catch((error) => {
+    console.log("Error in set timeout promise");
+  });
+
+async function getCurrentLocation() {
+  return new Promise((resolve) => {
+    navigator.geolocation.getCurrentPosition(function (success) {
```

Comment:
It's good practice to check if the `getCurrentPosition(...)` doesn't succeed. In that case you can use the `reject` and not `resolve` callback for the promise.

See example:
`let order = ( time, work ) => {

  return new Promise( ( resolve, reject )=>{

    if( is_shop_open ){

      resolve( )

    }

    else{

      reject( console.log("Our shop is closed") )

    }

  })
}
`

Reply:
Thank you for your feedback.